### PR TITLE
Fix the input parameters for the querying strategies

### PR DESCRIPTION
**Pull request by polinankoleva**
- add input parameter for the strategies (similar to one in the loader)
- remove minimum group size parameter
- add a parameter for using WPT without grouping the triples by subject
- update README file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,20 +80,23 @@ If you wish to do so, add the following lines to the log4j.properties file:
 ## PRoST-Query: Querying with SPARQL
 To query the data use the following command:
 
-    spark2-submit --class run.Main PRoST-Query.jar -i <SPARQL_query> -d <DB_name> -s <stats_file> -o <HDFS_output_file> -wpt
+    spark2-submit --class run.Main PRoST-Query.jar -i <SPARQL_query> -d <DB_name> -s <stats_file> -o <HDFS_output_file> -lp <logical_partition_strategies>
```
Comment (gschievelbein) [inline README.md:100]: This way is better.

File: query/PRoST-query/src/main/java/translator/Translator.java [hunk 2]
```diff
@@ -192,9 +196,10 @@ public Node buildTree(final List<Triple> triples, final List<Var> projectionVars
...
 					}
 				}
-				// create node
+
+				// create appropriate node
 				if (joinedGroups.get(biggestGroupKey).getIwptGroup().size()
-						+ joinedGroups.get(biggestGroupKey).getWptGroup().size() >= minimumGroupSize) {
+						+ joinedGroups.get(biggestGroupKey).getWptGroup().size() >= DEFAULT_MIN_GROUP_SIZE) {
 					nodesQueue.add(new JptNode(joinedGroups.get(biggestGroupKey), prefixes));
 				} else {
```
Comment (gschievelbein) [inline query/PRoST-query/src/main/java/translator/Translator.java]: It's creating VP nodes without checking if VP is enabled

**Review by gschievelbein — CHANGES_REQUESTED** [review]
The creation or not of VP nodes is now set as an input parameter. That's fine. During the creation of nodes, this is not always checked, so it is possible to create VP nodes even when VP is disabled. 

Since the minimum group size input parameter was removed, it is now not possible to use grouped strategies only. 

It makes more sense for the input for grouping to be something like enablesGrouping, since IWPT and JWPT do not group by subject only. This would also make it clearer why the minimum group size constraint is not always enforced.

**Comment by polinankoleva** [comment]
@arrascue I added some changes to the last pull, can you take a look and merge the request?

**Comment by polinankoleva** [comment]
@gschievelbein because u requested the changes, can you take a look at them and approve/disapprove them?

**Comment by gschievelbein** [comment]
Is there any specific reason why the minimum group size can't be changed anymore?

With constant minimum size, we have to make conditional tests for cases when we are not grouping triples, and cases when VP is not activated. We had many nested ifs already, I tried to reduce them with the refactoring.

If we minimum group size modifiable, the changes for the new input parameter to enable VP and grouping can be done with minimal changes.

The only changes necessary would be:

1. When reading the input parameters, set the minimum size to 1 if we are not grouping or we don't enable VP.

2. in the grouping methods, when grouping is disable, each triple pattern is in a group with only one element.

If we keep min group size modifiable, I can make these changes and push here.

**Comment by gschievelbein** [comment]
I did the changes I mentioned. 

Basically, when we group the triples, if grouping is disabled, each triples list will contain only one triple. The minimum group size is also set to 1, so the PT nodes can be created.

The "not grouping" of triples was not done for JWPT and IWPT models, so I did that too.

It needs testing. I haven't run it yet.

**Review by gschievelbein — APPROVED** [review]

**Pull request merged**